### ISSUE TITLE
Invalid option value "null" for rule after upgrading to 7.4.0

### ISSUE THREAD

**Issue by vkrol** [stylelint/stylelint#1964]
After upgrading to 7.4.0 stylelint shows the following warnings in the console:

```
Invalid Option: Invalid option value "null" for rule "{RULE NAME}-blacklist"
```

> Which rule, if any, is this issue related to?

function-blacklist
function-whitelist
unit-blacklist
unit-whitelist
property-blacklist
property-whitelist
selector-attribute-operator-blacklist
selector-attribute-operator-whitelist
selector-pseudo-class-blacklist
selector-pseudo-class-whitelist
at-rule-blacklist
at-rule-whitelist
comment-word-blacklist

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "function-whitelist": null
  }
}
```

> Which version of stylelint are you using?

`7.4.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

gulp-stylelint

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
Invalid Option: Invalid option value "null" for rule "function-blacklist"
```

**Comment by daltones** [comment]
I can confirm the issue after 7.4.0.

I also noticed that only `declaration-property-unit-{blacklist,whitelist}`, `declaration-property-value-{blacklist,whitelist}` and `media-feature-name-{blacklist,whitelist}` are okay with `null` values.

**Comment by vkrol** [comment]
@daltones 

> I also noticed that only declaration-property-unit-{blacklist,whitelist}, declaration-property-value-{blacklist,whitelist} and media-feature-name-{blacklist,whitelist} are okay with null values.

You are right, Sorry for the confusion. I edited the original comment.

**Comment by davidtheclark** [comment]
Sorry! This is a regression after some essential refactoring in 7.4.0. I will get a fix out promptly.

**Comment by jeddy3** [comment]
@vkrol & @daltones Thanks for reporting the issue. @davidtheclark's fix was released in `7.4.1`.

**Comment by davidtheclark** [comment]
Agh I just realized this is probably not the right solution! I am just starting a hike so will not write the next patch until this evening ... but I realized that the options validation should not even be being invoked for these nullified rules! Something earlier in the process should stop it. I suspect that there's just some line that should be looking for either null or an array containing null as the first value, and it's only doing the first.

If nobody looks into this first I will get to it tonight after my hike 😑

**Comment by jeddy3** [comment]
> but I realized that the options validation should not even be being invoked for these nullified rules! 

Oops, I should have caught this during the review.

I'm afraid I've got to crash now. As such, feel free to merge and release the next patch at your discretion.

Enjoy your hike! :)

**Comment by davidtheclark** [comment]
With v7.4.2 the rule code should not be run at all if the settings is `null`.
